### DISCUSSION TITLE
A little bit on Branches (both for caMicroscope and your forks)

### DISCUSSION THREAD

**Discussion by birm** [camicroscope/caMicroscope#919] [General]
We're working on keeping up to date with PRs, and noticed some duplication and conflicts; this mostly seems to be beause people are basing their PRs on master rather than develop. I know that caMicroscope does things in a less common way for an open source project, so hoping this helps.

I'd recommend that you don't push to master or develop on your fork. (Honestly, unless you're testing, you probably don't need to worry about master, so we'll just talk about develop when...)

Keeping develop clean means that you can keep it in line with upstream, and use your fork's develop branch to make your PR topic branch (that is, checkout develop, then make your branch based off of that)

To keep your develop up to date on occasion, add the main caMicroscope repo as upstream
`git remote add upstream https://github.com/camicroscope/camicroscope.git`
then you can bring in new changes from upstream by `git fetch upstream` then `git checkout develop
git merge upstream/develop` (and push to your fork's develop if you want)

We'll still try to get through open PRs as we can, but this should make it easier for everyone. Hope this helps!

And again, thanks for the overwhelming interest/support. We're behind because we underestimated the level of interest and support; it's truly amazing.

**Comment by gbengaoluwadahunsi** [comment]
Thank you very much @birm, 

It has been awesome working on this project and learning from your wealth of experience too.

**Comment by barchakuz** [comment]
Thank you @birm  for your assistance. It has been a valuable learning experience contributing here.

**Comment by Lochipi** [comment]
Sure, thanks.
